blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/JwtAuthenticationToken.java

```diff
@@ -0,0 +1,135 @@
+package io.jenkins.blueocean.service.embedded;
+
+import hudson.model.User;
+import io.jenkins.blueocean.auth.jwt.JwtToken;
+import io.jenkins.blueocean.commons.ServiceException;
+import jenkins.model.Jenkins;
+import org.acegisecurity.GrantedAuthority;
+import org.acegisecurity.GrantedAuthorityImpl;
+import org.acegisecurity.providers.AbstractAuthenticationToken;
+import org.acegisecurity.userdetails.UserDetails;
+import org.jose4j.jwt.JwtClaims;
+import org.jose4j.jwt.MalformedClaimException;
+import org.jose4j.jwt.NumericDate;
+import org.jose4j.jwt.consumer.InvalidJwtException;
+import org.jose4j.jwt.consumer.JwtConsumer;
+import org.jose4j.jwt.consumer.JwtConsumerBuilder;
+import org.jose4j.jwt.consumer.JwtContext;
+import org.jose4j.jwx.JsonWebStructure;
+import org.jose4j.lang.JoseException;
+import org.kohsuke.stapler.StaplerRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import static org.jose4j.jws.AlgorithmIdentifiers.RSA_USING_SHA256;
+
+/**
+ * @author Kohsuke Kawaguchi
+ */
+public final class JwtAuthenticationToken extends AbstractAuthenticationToken{
+
+    private static final Logger logger = LoggerFactory.getLogger(JwtAuthenticationToken.class);
+
+
+    private final String name;
+    private final GrantedAuthority[] grantedAuthorities;
+
+    public JwtAuthenticationToken(StaplerRequest request) {
+        String authHeader = request.getHeader("Authorization");
+        if(authHeader == null || !authHeader.startsWith("Bearer")){
```

Comment:
By right, should be checking that the header starts with "Bearer " i.e. including the space after the word.